src/MissionManager/ComplexMissionItem.cc

```diff
@@ -396,6 +396,24 @@ void ComplexMissionItem::_intersectLinesWithPolygon(const QList<QLineF>& lineLis
     }
 }
 
+/// Adjust the line segments such that they are all going the same direction with respect to going from P1->P2
+void ComplexMissionItem::_adjustLineDirection(const QList<QLineF>& lineList, QList<QLineF>& resultLines)
+{
+    for (int i=0; i<lineList.count(); i++) {
+        const QLineF& line = lineList[i];
+        QLineF adjustedLine;
+
+        if (line.angle() > 180.0) {
```

Comment:
I would expect that the flipped lines would have an angle of 180.0°, which means they wouldn't be flipped back.

And if you change it to >=, this statement restricts the choice of grid angles to [0° - 180°), as otherwise the segments are flipped. Instead I would propose to alter the check to something like
```cpp
if (fmod(fabs(line.angle() - gridAngle), 180.0) >= 90)
```